configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionFactory.java

```diff
@@ -161,23 +164,47 @@ private LocalSession create(File applicationFile, ApplicationId applicationId, l
                                 boolean internalRedeploy, TimeoutBudget timeoutBudget) {
         long sessionId = getNextSessionId();
         try {
-            ensureSessionPathDoesNotExist(sessionId);
+            ApplicationPackage app = createApplicationPackage(applicationFile, applicationId,
+                                                              sessionId, currentlyActiveSessionId, internalRedeploy);
+            return createSessionFromApplication(app, sessionId, timeoutBudget, clock);
+        } catch (Exception e) {
+            throw new RuntimeException("Error creating session " + sessionId, e);
+        }
+    }
+
+    /**
+     * This method is used when creating a session based on a remote session and the distributed application package
+     * It does not wait for session being created on other servers
```

Comment:
The other methods will do that because at that point the session is created for the first time and we want to wait until it is created on all servers. In this method we know that has already happened. Doing so would just be unnecessary (probably not an error, though, I haven't looked int it)